### ISSUE TITLE
TypeScript compilation fails

### ISSUE THREAD

**Issue by DKroot** [Treri/http-ng#1]
When I consume `http-ng` via TypeScript it fails:

`>npm install http-ng`

```
import IHttp from 'http-ng';

node_modules/http-ng/http.d.ts(171,19): error TS1003: Identifier expected.
node_modules/http-ng/http.d.ts(171,32): error TS1005: ')' expected.
node_modules/http-ng/http.d.ts(171,49): error TS1109: Expression expected.
node_modules/http-ng/http.d.ts(171,73): error TS1005: ';' expected.
node_modules/http-ng/http.d.ts(171,74): error TS1128: Declaration or statement expected.
node_modules/http-ng/http.d.ts(171,115): error TS1005: '(' expected.
```

**Comment by DKroot** [comment]
This is happening with the latest TypeScript: 2.3.4. It looks like `delete` is a keyword which it does not like. If I replace it with e.g. `doDelete`, compilation passes.

**Comment by Treri** [comment]
OK. I will have a look

**Comment by Treri** [comment]
Tanks for your report.

I published a new version v0.3.6

there is a new `del` method with old `delete` method reserved in js file.
But in type define file, only one `del` method
